app/src/main/java/org/mozilla/fenix/library/bookmarks/BookmarkView.kt

```diff
@@ -139,6 +139,8 @@ class BookmarkView(
         }
 
         bookmarkAdapter.updateData(state.tree, mode)
+        bookmarkAdapter.notifyDataSetChanged()
```

Comment:
Maybe we could move this up to only be called if the mode has changed?

Reply:
Yes, please check it now